test/src/org/omegat/gui/glossary/GlossarySearcherTest.java

```diff
@@ -0,0 +1,272 @@
+
+            public boolean isProjectLoaded() {
+                return true;
+            }
+
+            public boolean isOrphaned(EntryKey entry) {
+                return false;
+            }
+
+            public boolean isOrphaned(String source) {
+                return false;
+            }
+
+            public TMXEntry getTranslationInfo(SourceTextEntry ste) {
+                return null;
+            }
+
+            public AllTranslations getAllTranslations(SourceTextEntry ste) {
+                return null;
+            }
+
+            public Map<String, ExternalTMX> getTransMemories() {
+                return null;
+            }
+
+            public ITokenizer getTargetTokenizer() {
+                return null;
+            }
+
+            public StatisticsInfo getStatistics() {
+                return null;
+            }
+
+            public ITokenizer getSourceTokenizer() {
+                return null;
+            }
+
+            public ProjectProperties getProjectProperties() {
+                return props;
+            }
+
+            public List<IProject.FileInfo> getProjectFiles() {
+                return null;
+            }
+
+            public Map<Language, ProjectTMX> getOtherTargetLanguageTMs() {
+                return null;
+            }
+
+            public List<SourceTextEntry> getAllEntries() {
+                return null;
+            }
+
+            public void compileProject(String sourcePattern) {
+            }
+
+            public void closeProject() {
+            }
+
+            public List<String> getSourceFilesOrder() {
+                return null;
+            }
+
+            public void setSourceFilesOrder(List<String> filesList) {
+            }
+
+            @Override
+            public String getTargetPathForSourceFile(String sourceFile) {
+                return null;
+            }
+
+            @Override
+            public boolean isTeamSyncPrepared() {
+                return false;
+            }
+
+            @Override
+            public void teamSync() {
+            }
+
+            @Override
+            public void teamSyncPrepare() {
+            }
+
+            @Override
+            public boolean isRemoteProject() {
+                return false;
+            }
+
+            @Override
+            public void commitSourceFiles() {
+            }
+
+            @Override
+            public void compileProjectAndCommit(String sourcePattern, boolean doPostProcessing, boolean commitTargetFiles) {
+            }
+        });
+    }
+
+    private List<GlossaryEntry> glossarySearcherCommon(String sourceText, ITokenizer tok, Language language,
+                                                       List<GlossaryEntry> entries) {
+        EntryKey key = new EntryKey("file", sourceText, "id", "prev", "next", "path");
+        String[] prop = new String[]{};
+        SourceTextEntry ste = new SourceTextEntry(key, 1, prop, sourceText, new ArrayList<>());
```

Comment:
When you need an empty list, it's good to use `Collections#emptyList`.

Same goes for sets and maps.